src/cursor/abstract_cursor.ts

```diff
@@ -204,10 +204,22 @@ export abstract class AbstractCursor<
       ...pluckBSONSerializeOptions(options)
     };
     this.cursorOptions.timeoutMS = options.timeoutMS;
```

Comment:
(optional)  We have two patterns for attaching options from `options` into `cursorOptions`:

1. CSOT options are attached after the object is built
2. Readpreference is attached inline, with a conditional

Can we consolidate?

```typescript
    readPreference:
        options.readPreference && options.readPreference instanceof ReadPreference
          ? options.readPreference
          : ReadPreference.primary,
    ...pluckBSONSerializeOptions(options),
	timeoutMS: options.timeoutMS,
	tailable: options.tailable,
	awaitData: options.awaitData
```

(or the other way around would be fine too.)